htscodecs/rANS_static.c

```diff
@@ -682,28 +570,13 @@ unsigned char *rans_uncompress_O1(unsigned char *in, unsigned int in_size,
     unsigned int x;
     unsigned int out_sz, in_sz;
     char *out_buf = NULL;
-    // D[] is 1Mb and syms[][] is 0.5Mb.
-#ifndef NO_THREADS
-    pthread_once(&rans_once, rans_tls_init);
-    thread_data *tb = pthread_getspecific(rans_key);
-    if (!tb) {
-	if (!(tb = rans_tb_alloc()))
-	    return NULL;
-	pthread_setspecific(rans_key, tb);
-    }
-    ari_decoder *const D = tb->D;
-    RansDecSymbol32 (*const syms)[256] = tb->syms;
-#else
-#ifdef USE_HEAP
-    //ari_decoder *const D = malloc(256 * sizeof(*D));
-    ari_decoder *const D = calloc(256, sizeof(*D));
-    RansDecSymbol32 (*const syms)[256] = malloc(256 * sizeof(*syms));
-    for (i = 1; i < 256; i++) memset(&syms[i][0], 0, sizeof(syms[0][0]));
-#else
-    ari_decoder D[256] = {{{0}}}; //256*4k    => 1.0Mb
-    RansDecSymbol32 syms[256][256+6] = {{{0}}}; //256*262*8 => 0.5Mb
-#endif
-#endif
+    RansDecSymbol32 (*syms)[256];
+    uint8_t *mem = htscodecs_tls_calloc(256, sizeof(ari_decoder)
```

Comment:
There are a number of `return NULL;` statements below that could cause this memory to leak.  They either need to be changed to `goto cleanup;` or the checks involved should be moved above this allocation.